### PR TITLE
Add requirements hashes

### LINKED ISSUE #1457: Packages do not match the hashes from the requirements file
I'm unable to run Liberapay dev environment on *macOS Mojave*:

```
$ make clean env
...
Collecting yarl==1.3.0 (from -r requirements_tests.txt (line 112))
  Using cached https://files.pythonhosted.org/packages/fb/84/6d82f6be218c50b547aa29d0315e430cf8a23c52064c92d0a8377d7b7357/yarl-1.3.0.tar.gz
Collecting multidict==4.5.2 (from -r requirements_tests.txt (line 119))
  Using cached https://files.pythonhosted.org/packages/0e/64/39b167d63e292cc2336d09c4d3577089db82cd7886ed535b6100fc7c7966/multidict-4.5.2-cp37-cp37m-macosx_10_12_intel.macosx_10_12_x86_64.macosx_10_13_intel.macosx_10_13_x86_64.whl
Collecting fake-factory==0.3.2 (from -r requirements_tests.txt (line 136))
  Using cached https://files.pythonhosted.org/packages/de/38/748a54953a65fbb970d5619bef360819ab7f8cd5d3fbbb5092df03dde01f/fake-factory-0.3.2.tar.gz
Requirement already satisfied: setuptools in ./env/lib/python3.7/site-packages (from pytest==3.6.4->-r requirements_tests.txt (line 1)) (39.1.0)
THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    MarkupSafe==1.1.0 from https://files.pythonhosted.org/packages/96/52/eef455862764cb6d6c136aa52c7f9fc4e7c1c644790a7107b1244a2b8a53/MarkupSafe-1.1.0-cp37-cp37m-macosx_10_6_intel.whl#sha256=1f19ef5d3908110e1e891deefb5586aae1b49a7440db952454b4e281b41620cd (from -r requirements_base.txt (line 41)):
        Expected sha256 efdc45ef1afc238db84cb4963aa689c0408912a0239b0721cb172b4016eb31d6
        Expected     or 52ccb45e77a1085ec5461cde794e1aa037df79f473cbc69b974e73940655c8d7
        Expected     or 525396ee324ee2da82919f2ee9c9e73b012f23e7640131dd1b53a90206a0f09c
        Expected     or 5c3fbebd7de20ce93103cb3183b47671f2885307df4a17a0ad56a1dd51273d36
        Expected     or f82e347a72f955b7017a39708a3667f106e6ad4d10b25f237396a7115d8ed5fd
        Expected     or 31cbb1359e8c25f9f48e156e59e2eaad51cd5242c05ed18a8de6dbe85184e4b7
        Expected     or edce2ea7f3dfc981c4ddc97add8a61381d9642dc3273737e756517cc03e84dd6
        Expected     or bf54103892a83c64db58125b3f2a43df6d2cb2d28889f14c78519394feb41492
        Expected     or 048ef924c1623740e70204aa7143ec592504045ae4429b59c30054cb31e3c432
        Expected     or 83381342bfc22b3c8c06f2dd93a505413888694302de25add756254beee8449c
        Expected     or 130f844e7f5bdd8e9f3f42e7102ef1d49b2e6fdf0d7526df3f87281a532d8c8b
        Expected     or 52b07fbc32032c21ad4ab060fec137b76eb804c4b9a1c7c7dc562549306afad2
        Expected     or 4e97332c9ce444b0c2c38dd22ddc61c743eb208d916e4265a2a3b575bdccb1d3
             Got        1f19ef5d3908110e1e891deefb5586aae1b49a7440db952454b4e281b41620cd

    libsass==0.17.0 from https://files.pythonhosted.org/packages/a0/60/d1332c95d149d8e4976208e8bb6db39a0746b6ee46e94d840da3a0b05629/libsass-0.17.0-cp37-cp37m-macosx_10_14_x86_64.whl#sha256=1aeadc155594af23879e27792667dc06e7f248c2c599c40ff2a7335193abdf05 (from -r requirements_base.txt (line 125)):
        Expected sha256 bf6b7ad08f287695338f050c80f79d258a405e5c349cdaeb9be5d5376c09e37a
        Expected     or 0da943e00e028211cb4bb91496a20becab9fe82407bb75266ec4212af04acb45
        Expected     or 411833c623288138744865d882f5226f6db52afce1e19f42722c416df9d308dc
        Expected     or 66e3062ff508c81928e35c66702f0cc4f70fb12eb76ba23eeb0ff87a6340cc13
        Expected     or a19041e78d5bb7c5d72e010e893c29119693628b6ee06025503ab2584cf24edd
        Expected     or 107591ba2c0d173bb1705bef0e9fd04a5b6f482f3584f4ea51b28ab8b137fbb2
        Expected     or 4bf7a80a956da9de9715436b85343a179da4ff399a6e9a1694e70bff93d43099
        Expected     or 55b77204cfa363142ab02c49ee871321a396b8e51f6361ebc226c3953c780541
        Expected     or 953ebe810f09d81b84ccafdca0fb6171d1b58c8f0147cb650184a41e124e296f
             Got        1aeadc155594af23879e27792667dc06e7f248c2c599c40ff2a7335193abdf05

    coverage==4.5.2 from https://files.pythonhosted.org/packages/be/88/7e5e548329eda1f003b3ff34e57ba6b2b1f8b8983043e99a0ecf58ae0a06/coverage-4.5.2-cp37-cp37m-macosx_10_13_x86_64.whl#sha256=d64b4340a0c488a9e79b66ec9f9d77d02b99b772c8b8afd46c1294c1d39ca478 (from -r requirements_tests.txt (line 35)):
        Expected sha256 b3b0c8f660fae65eac74fbf003f3103769b90012ae7a460863010539bb7a80da
        Expected     or 8cb4febad0f0b26c6f62e1628f2053954ad2c555d67660f28dfb1b0496711952
        Expected     or 447c450a093766744ab53bf1e7063ec82866f27bcb4f4c907da25ad293bba7e3
        Expected     or 1b4276550b86caa60606bd3572b52769860a81a70754a54acc8ba789ce74d607
        Expected     or 6694d5573e7790a0e8d3d177d7a416ca5f5c150742ee703f3c18df76260de794
        Expected     or 510986f9a280cd05189b42eee2b69fecdf5bf9651d4cd315ea21d24a964a3c36
        Expected     or 09e47c529ff77bf042ecfe858fb55c3e3eb97aac2c87f0349ab5a7efd6b3939f
        Expected     or 5535dda5739257effef56e49a1c51c71f1d37a6e5607bb25a5eee507c59580d1
        Expected     or 0cc941b37b8c2ececfed341444a456912e740ecf515d560de58b9a76562d966d
        Expected     or da969da069a82bbb5300b59161d8d7c8d423bc4ccd3b410a9b4d8932aeefc14b
        Expected     or 6831e1ac20ac52634da606b658b0b2712d26984999c9d93f0c6e59fe62ca741b
        Expected     or 5f55028169ef85e1fa8e4b8b1b91c0b3b0fa3297c4fb22990d46ff01d22c2d6c
        Expected     or 10e8af18d1315de936d67775d3a814cc81d0747a1a0312d84e27ae5610e313b0
        Expected     or aaa0f296e503cda4bc07566f592cd7a28779d433f3a23c48082af425d6d5a78f
        Expected     or 1e8a2627c48266c7b813975335cfdea58c706fe36f607c97d9392e61502dc79d
        Expected     or 46101fc20c6f6568561cdd15a54018bb42980954b79aa46da8ae6f008066a30e
        Expected     or ee5b8abc35b549012e03a7b1e86c09491457dba6c94112a2482b18589cc2bdb9
        Expected     or c45297bbdbc8bb79b02cf41417d63352b70bcb76f1bbb1ee7d47b3e89e42f95d
        Expected     or ab235d9fe64833f12d1334d29b558aacedfbca2356dfb9691f2d0d38a8a7bfb4
        Expected     or 99bd767c49c775b79fdcd2eabff405f1063d9d959039c0bdd720527a7738748a
        Expected     or 71afc1f5cd72ab97330126b566bbf4e8661aab7449f08895d21a5d08c6b051ff
        Expected     or 7349c27128334f787ae63ab49d90bf6d47c7288c63a0a5dfaa319d4b4541dd2c
        Expected     or 859714036274a75e6e57c7bab0c47a4602d2a8cfaaa33bbdb68c8359b2ed4f5c
        Expected     or 0d34245f824cc3140150ab7848d08b7e2ba67ada959d77619c986f2062e1f0e8
        Expected     or 3ad59c84c502cd134b0088ca9038d100e8fb5081bbd5ccca4863f3804d81f61d
             Got        d64b4340a0c488a9e79b66ec9f9d77d02b99b772c8b8afd46c1294c1d39ca478

    multidict==4.5.2 from https://files.pythonhosted.org/packages/0e/64/39b167d63e292cc2336d09c4d3577089db82cd7886ed535b6100fc7c7966/multidict-4.5.2-cp37-cp37m-macosx_10_12_intel.macosx_10_12_x86_64.macosx_10_13_intel.macosx_10_13_x86_64.whl#sha256=1d48bc124a6b7a55006d97917f695effa9725d05abe8ee78fd60d6588b8344cd (from -r requirements_tests.txt (line 119)):
        Expected sha256 068167c2d7bbeebd359665ac4fff756be5ffac9cda02375b5c5a7c4777038e73
        Expected     or 7c1b7eab7a49aa96f3db1f716f0113a8a2e93c7375dd3d5d21c4941f1405c9c5
        Expected     or 8ccd1c5fff1aa1427100ce188557fc31f1e0a383ad8ec42c559aabd4ff08802d
        Expected     or 6a3a9b0f45fd75dc05d8e93dc21b18fc1670135ec9544d1ad4acbcf6b86781d0
        Expected     or 31dfa2fc323097f8ad7acd41aa38d7c614dd1960ac6681745b6da124093dc351
        Expected     or 8e08dd76de80539d613654915a2f5196dbccc67448df291e69a88712ea21e24a
        Expected     or d1071414dd06ca2eafa90c85a079169bfeb0e5f57fd0b45d44c092546fcd6fd9
        Expected     or db603a1c235d110c860d5f39988ebc8218ee028f07a7cbc056ba6424372ca31b
        Expected     or ce20044d0317649ddbb4e54dab3c1bcc7483c78c27d3f58ab3d0c7e6bc60d26a
        Expected     or 4b843f8e1dd6a3195679d9838eb4670222e8b8d01bc36c9894d6c3538316fa0a
        Expected     or 047c0a04e382ef8bd74b0de01407e8d8632d7d1b4db6f2561106af812a68741b
        Expected     or 148ff60e0fffa2f5fad2eb25aae7bef23d8f3b8bdaf947a65cdbe84a978092bc
        Expected     or 4b02a3b2a2f01d0490dd39321c74273fed0568568ea0e7ea23e02bd1fb10a10b
        Expected     or d3be11ac43ab1a3e979dac80843b42226d5d3cccd3986f2e03152720a4297cd7
        Expected     or 024b8129695a952ebd93373e45b5d341dbb87c17ce49637b34000093f243dd4f
             Got        1d48bc124a6b7a55006d97917f695effa9725d05abe8ee78fd60d6588b8344cd
```

It seems quite related to https://github.com/liberapay/liberapay.com/issues/1435.

@mattbk Did it solved your issue?

For information, my python installed version is 3.7.2.

### PR DESCRIPTION
**Pull request by MartinDelille**
This PR try to fix #1457 by adding the required hashes.

I let it has a draft because I still have a problem to get the environnement up and running.

When running the following command it fails:

```
$ make clean env
...
    clang -bundle -undefined dynamic_lookup -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk build/temp.macosx-10.14-x86_64-3.7/psycopg/psycopgmodule.o build/temp.macosx-10.14-x86_64-3.7/psycopg/green.o build/temp.macosx-10.14-x86_64-3.7/psycopg/pqpath.o build/temp.macosx-10.14-x86_64-3.7/psycopg/utils.o build/temp.macosx-10.14-x86_64-3.7/psycopg/bytes_format.o build/temp.macosx-10.14-x86_64-3.7/psycopg/libpq_support.o build/temp.macosx-10.14-x86_64-3.7/psycopg/win32_support.o build/temp.macosx-10.14-x86_64-3.7/psycopg/connection_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/connection_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/cursor_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/cursor_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_connection_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_cursor_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_message_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/diagnostics_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/error_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/lobject_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/lobject_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/notify_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/xid_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_asis.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_binary.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_datetime.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_list.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pboolean.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pdecimal.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pint.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pfloat.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_qstring.o build/temp.macosx-10.14-x86_64-3.7/psycopg/microprotocols.o build/temp.macosx-10.14-x86_64-3.7/psycopg/microprotocols_proto.o build/temp.macosx-10.14-x86_64-3.7/psycopg/typecast.o -L/usr/local/lib -lpq -lssl -lcrypto -o build/lib.macosx-10.14-x86_64-3.7/psycopg2/_psycopg.cpython-37m-darwin.so
    ld: library not found for -lssl
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: command 'clang' failed with exit status 1

    ----------------------------------------
Command "/Users/martin/dev/liberapay/liberapay.com/env/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-install-s7y94_gg/psycopg2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-record-hdjsu37j/install-record.txt --single-version-externally-managed --compile --install-headers /Users/martin/dev/liberapay/liberapay.com/env/bin/../include/site/python3.7/psycopg2" failed with error code 1 in /private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-install-s7y94_gg/psycopg2/
```

I'm on *macOS 10.14.3*.

### CODE REVIEW

**Comment by MartinDelille** [comment]
(I didn't made a draft by mistake)

**Comment by Changaco** [comment]
It looks to me like you've only added the hashes of the packages that your version of pip downloaded. That doesn't tell us which files these hashes correspond to, why they weren't already included, nor how we can prevent this issue from happening again.

**Comment by MartinDelille** [comment]
Of course but this is the only way I found to pass this verification step.

Where does this list come from? https://github.com/liberapay/liberapay.com/blob/9398edc5805677115dc8a30c57e852bbf024b1be/requirements_base.txt#L42-L55

**Comment by Changaco** [comment]
The hashes are generated by [hashin](https://github.com/peterbe/hashin), as mentioned in <https://github.com/liberapay/liberapay.com#modifying-python-dependencies>.

**Comment by MartinDelille** [comment]
After investigating I noticed the `hashin` check wasn't handling 3.7 package which explains why the hash was not contained in the requirement lists.

I replaced the `sed -r` by `sed -E` in the `Makefile` which is equivalent (https://www.gnu.org/software/sed/manual/sed.html#Command_002dLine-Options) but `sed -r` doesn't work on my mac.

I tried running `make rehash-requirements` but there is some issue on my side:
- it trigger an error for lines with comment like here: https://github.com/liberapay/liberapay.com/blob/master/requirements_base.txt#L9
- it fails on these two lines:
  - https://github.com/liberapay/liberapay.com/blob/master/requirements_tests.txt#L112
  - https://github.com/liberapay/liberapay.com/blob/master/requirements_tests.txt#L119
- `hashin` doesn't seem to handle well indented requirement files

Good news: my environment is up and running now! 😄

**Comment by MartinDelille** [comment]
I finally still had the problem with my macbook on 10.14:

```
$ make clean env
...
    clang -bundle -undefined dynamic_lookup -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk build/temp.macosx-10.14-x86_64-3.7/psycopg/psycopgmodule.o build/temp.macosx-10.14-x86_64-3.7/psycopg/green.o build/temp.macosx-10.14-x86_64-3.7/psycopg/pqpath.o build/temp.macosx-10.14-x86_64-3.7/psycopg/utils.o build/temp.macosx-10.14-x86_64-3.7/psycopg/bytes_format.o build/temp.macosx-10.14-x86_64-3.7/psycopg/libpq_support.o build/temp.macosx-10.14-x86_64-3.7/psycopg/win32_support.o build/temp.macosx-10.14-x86_64-3.7/psycopg/connection_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/connection_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/cursor_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/cursor_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_connection_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_cursor_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/replication_message_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/diagnostics_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/error_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/lobject_int.o build/temp.macosx-10.14-x86_64-3.7/psycopg/lobject_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/notify_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/xid_type.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_asis.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_binary.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_datetime.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_list.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pboolean.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pdecimal.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pint.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_pfloat.o build/temp.macosx-10.14-x86_64-3.7/psycopg/adapter_qstring.o build/temp.macosx-10.14-x86_64-3.7/psycopg/microprotocols.o build/temp.macosx-10.14-x86_64-3.7/psycopg/microprotocols_proto.o build/temp.macosx-10.14-x86_64-3.7/psycopg/typecast.o -L/usr/local/lib -lpq -lssl -lcrypto -o build/lib.macosx-10.14-x86_64-3.7/psycopg2/_psycopg.cpython-37m-darwin.so
    ld: library not found for -lssl
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: command 'clang' failed with exit status 1

    ----------------------------------------
Command "/Users/martin/dev/liberapay/liberapay.com/env/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-install-s7y94_gg/psycopg2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-record-hdjsu37j/install-record.txt --single-version-externally-managed --compile --install-headers /Users/martin/dev/liberapay/liberapay.com/env/bin/../include/site/python3.7/psycopg2" failed with error code 1 in /private/var/folders/rv/x9flf_k97zq_bfv790jhhtwc0000gn/T/pip-install-s7y94_gg/psycopg2/
```

I managed to fix it by upgrading psycopg2 to 2.7.7.

Do you think it is an acceptable solution?

If you're ok I might update all the requirement hashes including python 3.7.

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,10 @@ matrix:
   include:
     - python: 3.6
       env: TOXENV=py36
+    - python: 3.7
+      env: TOXENV=py37
+      dist: xenial
+      sudo: true
```
Comment (Changaco) [inline .travis.yml]: Why `sudo: true` here? Also, why `dist: xenial` only for Python 3.7 instead of globally?

Comment (MartinDelille) [reply .travis.yml]: It looks like this is the configuration required for python 3.7: https://github.com/travis-ci/travis-ci/issues/9069#issuecomment-425720905

I can set this configuration to global if you want.

**Pull request merged**